### DISCUSSION TITLE
add_row outside process_node/way/relation() function

### DISCUSSION THREAD

**Discussion by SteSad** [osm2pgsql-dev/osm2pgsql#1667] [Q&A] [answered]
In order to be able to react to changes in the database, I would like to store the time of use of osm2pgsql in a table.

I imagined it like this:

i would add a new table definition in the lua script, like:
```lua
tables.creationinformation = osm2pgsql.define_table({
    name = 'creationinformation',
    columns = {
        { column = 'id', sql_type = 'bigserial', create_only = true },
        { column = 'timestamp', type = 'text' }
    }
})
```

At the beginning of the conversion process, an entry is to be created in the creationinformation table, such as:
```lua
tables.creationinformation:add_row({
timestampstring = os.date('%Y-%m-%d %X',os.time())
}) 
```
___

### Problem
I cannot run the `add_row()` function outside of the `process_node/way/relation()` functions<br>
and I want to add the timestamp only once, not several times

### Question
Can anyone help me, or is there another way to realise my idea?

> **NOTE** I know I can do this with simple sql, but I want to do this in flex-output with lua

**Comment by joto** [comment, marked as answer]
You could do the `add_row()` in one of the process functions and set a global Lua variable when you have done so. But you shouldn't do tricks like that. For one I am not sure whether defining a table without id columns will work and, if it doesn't, you end up with a table with some kind of OSM id column and every time osm2pgsql sees a change it will try to delete the feature from that table, which adds more overhead.

If you insist in doing this from Lua, you can use the Lua Postgresql package do access the database from there. That would be a cleaner solution at least.
